tensorflow/compiler/mlir/hlo/tests/disc_ral_legalize_to_llvm.mlir

```diff
@@ -0,0 +1,144 @@
+// RUN: mlir-hlo-opt -disc-ral-to-llvm %s -o - | FileCheck %s
+
+// check global definitions
+// CHECK-DAG: llvm.mlir.global internal constant @ral_send_output___cpu___pvoid_i64_m2df32___void("ral_send_output___cpu___pvoid_i64_m2df32___void\00")
+// CHECK-DAG: llvm.mlir.global internal constant @ral_kernel_launch___cpu___pvoid_pvoid_pvoid_i64_i64_i64_i64_i64_i64_i32_i32_ppvoid___void("ral_kernel_launch___cpu___pvoid_pvoid_pvoid_i64_i64_i64_i64_i64_i64_i32_i32_ppvoid___void\00")
+// CHECK-DAG: llvm.mlir.global internal constant @kernel_module_the_kernel_kernel_name("the_kernel\00")
+// CHECK-DAG: llvm.mlir.global internal constant @kernel_module_blob("BLOB!")
+// CHECK-DAG: llvm.mlir.global internal constant @ral_recv_input___cpu___pvoid_i64___m2df32("ral_recv_input___cpu___pvoid_i64___m2df32\00")
+// CHECK-DAG: llvm.func @disc_ral_call(!llvm.ptr<i8>, !llvm.ptr<i8>, !llvm.ptr<ptr<i8>>)
+
+module @main attributes {gpu.container_module}  {
+  // CHECK-NOT: gpu.module
+  gpu.module @kernel_module attributes {gpu.binary_blob = "BLOB!"} {
+    llvm.func @the_kernel() attributes {gpu.kernel} {
+      llvm.return
+    }
+  }
```

Comment:
It seems like a precondition of this pass is that we already have a `gpu.binary_blob` produced. It seems to me that in such cases the module shouldn't also have an IR content?
Not only they seem redundant, but also they could get out-of-sync.

Reply:
Thanks! It does not need to contain the actual IR while we still need to keep the function declaration inside the module since the `gpu.launch_op` needs to refer it. Thus, I'll remove the content of the `the_kernel ` while keeping the declaration.